### PR TITLE
TimerCounter with logic + PR changes

### PR DESCRIPTION
**Pull request by mateuszpmroz**

### CODE REVIEW

File: src/screens/PinFlow/CurrentPinScreen.tsx [hunk 1]
```diff
@@ -14,37 +25,95 @@ interface Props {
...
     error: '',
+    failedTimes: 0,
+    isCount: true,
+  };
+
+  handleFailedAttempt = (increasedFailedTimes: number) => {
+    const isFinalAttempt = increasedFailedTimes > 2;
+    if (isFinalAttempt) {
+      const { attempt } = this.props.timeCounter;
+      let timestamp = dayjs();
```
Comment (KedziaPawel) [inline src/screens/PinFlow/CurrentPinScreen.tsx:52]: it's not timestamp

File: src/screens/PinFlow/CurrentPinScreen.tsx [hunk 2]
```diff
@@ -14,37 +25,95 @@ interface Props {
...
+    let blockedTimeInMinutes = 1;
+    if (attempt === 0) {
+      blockedTimeInMinutes = 1;
+    } else if (attempt === 1) {
+      blockedTimeInMinutes = 2;
+    } else if (attempt > 1) {
+      blockedTimeInMinutes = 10;
+    }
+    return increasedFailedTimes !== firstAttempt && increasedFailedTimes !== finalAttempt
+      ? `\n${i18n.onboarding.failedTimesErrorInfo} ${blockedTimeInMinutes} ${i18n.onboarding.minutes}\n${i18n.onboarding.failedTimes} ${increasedFailedTimes}/3`
```
Comment (KedziaPawel) [inline src/screens/PinFlow/CurrentPinScreen.tsx:81]: /${finalAttempt} at the end i guess

File: src/screens/PinFlow/CurrentPinScreen.tsx [hunk 3]
```diff
@@ -14,37 +25,95 @@ interface Props {
...
+    isCount: true,
+  };
+
+  handleFailedAttempt = (increasedFailedTimes: number) => {
+    const isFinalAttempt = increasedFailedTimes > 2;
+    if (isFinalAttempt) {
+      const { attempt } = this.props.timeCounter;
+      let timestamp = dayjs();
+
+      if (attempt === 0) {
```
Comment (KedziaPawel) [inline src/screens/PinFlow/CurrentPinScreen.tsx:54]: so this is duplicated logic with below, you are checking the same else ifs for atttempts, merge those two.
if (attempt === 0) {
      blockedTimeInMinutes = 1;
    } else if (attempt === 1) {
      blockedTimeInMinutes = 2;
    } else if (attempt > 1) {
      blockedTimeInMinutes = 10;
    }
    timestamp = timestamp.add(blockedTimeInMinutes, 'minute');

File: src/screens/PinFlow/CurrentPinScreen.tsx [hunk 4]
```diff
@@ -14,37 +25,95 @@ interface Props {
...
   };
+  setTimeCounter: (timestamp: number) => SetTimeCounterAction;
+  setFailedAttempts: (attempt: number) => SetFailedAttemptsAction;
+  timeCounter: TimeCounterState;
 }
 
 interface State {
   pin: string;
   error: string;
+  failedTimes: number;
```
Comment (KedziaPawel) [inline src/screens/PinFlow/CurrentPinScreen.tsx:36]: as I see you don't need local failedTimes as you already have this in redux. Reading from two sources of truth  makes it hard to debug and read the code

Comment (mateuszpmroz) [reply src/screens/PinFlow/CurrentPinScreen.tsx:36]: failedTimes and attempts are two different thigs, failedTimes are included in one attempt. So you have attempts and attempt could have maximum three failedTimes, then the timer is opening, attempt is increasing and failedTimes are reseting.

Comment (mateuszpmroz) [reply src/screens/PinFlow/CurrentPinScreen.tsx:36]: I will merge it due to testing process. If you have solution to make it more redable then write it there and I'm gonna apply it in other PR.

File: src/screens/PinFlow/CurrentPinScreen.tsx [hunk 5]
```diff
@@ -14,37 +25,87 @@ interface Props {
...
     this.setState({ pin }, async () => {
       if (this.state.pin.length === CONST.pinCodeLength) {
         const setPin = await SecureStorageService.getSecuredValue('pin');
         if (setPin === this.state.pin) {
+          setFailedAttempts(0);
           this.props.navigation.navigate(Route.CreatePin, {
             flowType: FlowType.newPin,
           });
         } else {
+          const increasedFailedTimes = this.state.failedTimes + 1;
```
Comment (KedziaPawel) [inline src/screens/PinFlow/CurrentPinScreen.tsx:89]: read this from redux and you don't need failedTimes locally

Comment (mateuszpmroz) [reply src/screens/PinFlow/CurrentPinScreen.tsx:89]: > read this from redux and you don't need failedTimes locally
@KedziaPawel 
That is not stored in redux, as I've written before:
`failedTimes and attempts are two different thigs, failedTimes are included in one attempt. So you have attempts and attempt could have maximum three failedTimes, then the timer is opening, attempt is increasing and failedTimes are reseting.`

Comment (KedziaPawel) [reply src/screens/PinFlow/CurrentPinScreen.tsx:89]: but this is the the same number  `const increasedFailedTimes = this.props. timeCounter.attempt + 1;` and it should work

**Pull request merged**